app/src/test/report/components/Reports/ReportPageTests.tsx

```diff
@@ -1,85 +1,59 @@
-            expect(fetchVersions.called).to.equal(true, "Expected fetchVersions to be called");
-            expect(fetchVersionDetails.called).to.equal(true, "Expected fetchVersionDetails to be called");
-            expectOrderedActions(spy, [
-                {action: "ReportActions.setCurrentReport", payload: "reportname"},
-                {action: "ReportActions.setCurrentVersion", payload: "versionname"}
-            ]);
-        });
-    };
 
     it("triggers actions on mount", (done: DoneCallback) => {
-        checkExpectedActionsWhen(done, () => {
-            return new ReportPage({location: location, router: null}).load({
-                report: "reportname",
-                version: "versionname"
-            });
+        const store = createMockStore({});
+
+        const location = mockLocation<ReportPageProps>({report: "report", version: "v1"});
+        sandbox.setStubFunc(ReportPageComponent.prototype, "getLocationParams", () => (location));
+        sandbox.setStubFunc(ReportPageComponent.prototype, "render", () => (<p/>));
+        const createBreadCrumbSpy = sandbox.setStub(ReportPageComponent.prototype, "createBreadcrumb");
+        sandbox.setStub(ReportsService.prototype, "getReportVersions");
+        sandbox.setStub(ReportsService.prototype, "getVersionDetails");
+
+        const rendered = mount(<Provider store={store}><ReportPage /></Provider>);
+        setTimeout(() => {
+            const actions = store.getActions();
+            expect(actions[0].type).to.eql(ReportTypeKeys.SET_CURRENT_REPORT);
+            expect(actions[1].type).to.eql(ReportTypeKeys.SET_CURRENT_VERSION);
+            expect(actions[2].type).to.eql(ReportTypeKeys.REPORT_VERSIONS_FETCHED);
+            expect(actions[3].type).to.eql(ReportTypeKeys.REPORT_VERSION_DETAILS_FETCHED);
+            expect(createBreadCrumbSpy.called).to.eq(true);
+            done();
         });
     });
 
     it("triggers actions and redirect when child triggers changeVersion", (done: DoneCallback) => {
         const redirectTo = sandbox.sinon.stub();
         const router: IRouter = {redirectTo};
-
-        checkExpectedActionsWhen(done, () => {
-            const location = mockLocation<ReportPageProps>({report: "reportname", version: "oldVersion"});
-            const page = new ReportPage({location: location, router: router});
-            const promise = page.changeVersion("versionname")
+        sandbox.setStub(ReportPageComponent.prototype, "loadVersion");
+        const location = mockLocation<ReportPageProps>({report: "reportname", version: "oldVersion"});
+        const page = new ReportPageComponent({location: location, router: router});
+        page.changeVersion("versionname")
+        setTimeout(() => {
             expect(redirectTo.called).to.equal(true, "Expected redirectTo to be called");
             expect(redirectTo.calledWith("/reportname/versionname/"));
-            return promise;
+            done();
```

Comment:
So the previous version of this test also tested whether the actions (set current report, fetch version details etc) were triggered - would be good to keep that in the new test.